lib/Containers/Helpers.h

```diff
@@ -33,28 +33,37 @@ namespace arangodb::containers {
 struct Helpers {
 /// @brief calculate capacity for the container for at least one more 
 /// element.
-/// if this would exceed the container's capacity, use a factor-of-2
-/// growth strategy to calculate the capacity.
+/// if this would exceed the container's capacity, use a growth factor of
+/// 1.5 to calculate the new capacity.
 template<typename T>
 static std::size_t nextCapacity(T const& container, std::size_t initialCapacity) {
   std::size_t capacity;
   if (container.empty()) {
     // reserve some initial space
     capacity = std::max(std::size_t(1), initialCapacity);
   } else {
+    TRI_ASSERT(container.capacity() > 0);
+    // minimum requirement is that we have room for at least one more element.
     capacity = container.size() + 1;
-    // allocate with power of 2 growth
     if (capacity > container.capacity()) {
-      capacity *= 2;
+      // inspired by facebook/folly (https://github.com/facebook/folly/blob/master/folly/memory/Malloc.h):
+      constexpr size_t jemallocMinInPlaceExpandable = 4096;
+      if (container.capacity() < jemallocMinInPlaceExpandable / sizeof(typename T::value_type)) {
```

Comment:
Would this be more accurate?
```suggestion
      if (container.capacity() < jemallocMinInPlaceExpandable / std::max(sizeof(typename T::value_type), alignof(typename T::value_type))) {
```